pool/swap.gno

```diff
@@ -0,0 +1,513 @@
+	state := comp.InitialState
+	swapFee := u256.Zero()
+
+
+	var newFee *u256.Uint
+	var err error
+
+	// Compute swap steps until completion
+	for shouldContinueSwap(state, comp.SqrtPriceLimitX96) {
+		state, newFee, err = computeSwapStep(state, pool, comp.ZeroForOne, comp.SqrtPriceLimitX96, comp.ExactInput, comp.Cache, swapFee)
+		if err != nil {
+			return nil, err
+		}
+		swapFee = newFee
+	}
+
+	// Calculate final amounts
+	amount0 := state.amountCalculated
+	amount1 := i256.Zero().Sub(comp.AmountSpecified, state.amountSpecifiedRemaining)
+	if comp.ZeroForOne == comp.ExactInput {
+		amount0, amount1 = amount1, amount0
+	}
+
+	// Prepare result
+	result := &SwapResult{
+		Amount0:      amount0,
+		Amount1:      amount1,
+		NewSqrtPrice: state.sqrtPriceX96,
+		NewTick:      state.tick,
+		NewLiquidity: state.liquidity,
+		NewProtocolFees: ProtocolFees{
+			token0: pool.protocolFees.token0,
+			token1: pool.protocolFees.token1,
+		},
+		FeeGrowthGlobal0X128: pool.feeGrowthGlobal0X128,
+		FeeGrowthGlobal1X128: pool.feeGrowthGlobal1X128,
+		SwapFee:          swapFee,
+	}
+
+	// Update protocol fees if necessary
+	if comp.ZeroForOne {
+		if state.protocolFee.Gt(u256.Zero()) {
+			result.NewProtocolFees.token0 = new(u256.Uint).Add(result.NewProtocolFees.token0, state.protocolFee)
+		}
+		result.FeeGrowthGlobal0X128 = state.feeGrowthGlobalX128
+	} else {
+		if state.protocolFee.Gt(u256.Zero()) {
+			result.NewProtocolFees.token1 = new(u256.Uint).Add(result.NewProtocolFees.token1, state.protocolFee)
+		}
+		result.FeeGrowthGlobal1X128 = state.feeGrowthGlobalX128
+	}
+
+	return result, nil
+}
+
+// applySwapResult updates pool state with computed results.
+// All state changes are applied at once to maintain consistency
+func applySwapResult(pool *Pool, result *SwapResult) {
+	pool.slot0.sqrtPriceX96 = result.NewSqrtPrice
+	pool.slot0.tick = result.NewTick
+	pool.liquidity = result.NewLiquidity
+	pool.protocolFees = result.NewProtocolFees
+	pool.feeGrowthGlobal0X128 = result.FeeGrowthGlobal0X128
+	pool.feeGrowthGlobal1X128 = result.FeeGrowthGlobal1X128
+}
+
+// validatePriceLimits ensures the provided price limit is valid for the swap direction
+// The function enforces that:
+// For zeroForOne (selling token0):
+//   - Price limit must be below current price
+//   - Price limit must be above MIN_SQRT_RATIO
+// For !zeroForOne (selling token1):
+//   - Price limit must be above current price
+//   - Price limit must be below MAX_SQRT_RATIO
+func validatePriceLimits(pool *Pool, zeroForOne bool, sqrtPriceLimitX96 *u256.Uint) {
+	if zeroForOne {
+		minSqrtRatio := u256.MustFromDecimal(consts.MIN_SQRT_RATIO)
+
+		cond1 := sqrtPriceLimitX96.Lt(pool.slot0.sqrtPriceX96)
+		cond2 := sqrtPriceLimitX96.Gt(minSqrtRatio)
+		if !(cond1 && cond2) {
+			panic(addDetailToError(
+				errPriceOutOfRange,
+				ufmt.Sprintf("pool.gno__Swap() || sqrtPriceLimitX96(%s) < slot0Start.sqrtPriceX96(%s) && sqrtPriceLimitX96(%s) > consts.MIN_SQRT_RATIO(%s)",
```

Comment:
```suggestion
				ufmt.Sprintf("pool.gno__validatePriceLimits() || sqrtPriceLimitX96(%s) < slot0Start.sqrtPriceX96(%s) && sqrtPriceLimitX96(%s) > consts.MIN_SQRT_RATIO(%s)",
```

Reply:
@notJoon @r3v4s 
Remove the paths from the error logs